sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionResolution.scala

```diff
@@ -128,18 +128,14 @@ class FunctionResolution(
       numArgs: Int,
       u: UnresolvedFunction): Expression = {
     func match {
-      case owg: SupportsOrderingWithinGroup if u.isDistinct =>
-        throw QueryCompilationErrors.distinctInverseDistributionFunctionUnsupportedError(
-          owg.prettyName
-        )
-      case owg: SupportsOrderingWithinGroup
-          if !owg.orderingFilled && u.orderingWithinGroup.isEmpty =>
-        throw QueryCompilationErrors.inverseDistributionFunctionMissingWithinGroupError(
-          owg.prettyName
-        )
+      case owg: SupportsOrderingWithinGroup if !owg.isDistinctSupported && u.isDistinct =>
+        throw QueryCompilationErrors.distinctWithOrderingFunctionUnsupportedError(owg.prettyName)
       case owg: SupportsOrderingWithinGroup
-          if owg.orderingFilled && u.orderingWithinGroup.nonEmpty =>
-        throw QueryCompilationErrors.wrongNumOrderingsForInverseDistributionFunctionError(
+          if owg.isOrderingMandatory && !owg.orderingFilled && u.orderingWithinGroup.isEmpty =>
+        throw QueryCompilationErrors.functionMissingWithinGroupError(owg.prettyName)
+      case owg: Mode if owg.orderingFilled && u.orderingWithinGroup.nonEmpty =>
```

Comment:
so only `Mode` function can make `orderingFilled` return true without `withOrderingWithinGroup` being called. I agree that this code change is harmless, but isn't it more general to match `SupportsOrderingWithinGroup` here? In case there are more function like `Mode` in the future.

Reply:
Maybe, but I don't like this behavior. It basically says "If a function got the order in some non-standard way then forbid `WITHIN GROUP (ORDER BY)` for it". I would prefer not to have functions that can bypass standard syntax. From my perspective, it's more a functions-specific hack than a part of the `SupportsOrderingWithinGroup` interface

Reply:
> "If a function got the order in some non-standard way then forbid WITHIN GROUP (ORDER BY) for it"

Isn't it a good behavior? We should not encourage new `SupportsOrderingWithinGroup` functions to get ordering in non-standard ways, and this check can enforce it. And this check was general before, please not change it.